packages/services/src/infrastructure/api/coincap/coincap-api.utils.ts

```diff
@@ -0,0 +1,23 @@
+import { CryptoCurrency } from '@leather.io/models';
+
+import { CoincapAssetResponse } from './coincap-api.client';
+
+export const coincapCurrencyMap: Record<CryptoCurrency, string> = {
+  BTC: 'bitcoin',
+  STX: 'stacks',
+};
+
+export function mapToCoincapCurrency(currency: CryptoCurrency): string {
+  return coincapCurrencyMap[currency];
+}
+
+export function readCoincapAssetPrice(assetResponse: CoincapAssetResponse): number | undefined {
+  try {
+    const priceUsd = assetResponse.data.priceUsd;
+    const price = parseFloat(priceUsd);
+    if (isNaN(price)) return;
```

Comment:
Should we throw errors for `undefined` data / add analytics? 

It could be a bit overkill for this type of thing but I'm curious to know your thoughts

Reply:
That's a great point. This utility is trying to handle bad responses, which isn't its responsibility. It should expect that the response object is what it says it is and any errors should be thrown upstream of this.

We should probably have Zod validation for all API response types, so that schema failures are detected immediately. Then we could try to centralize analytics directly within Zod to track validation errors. Looks like they support a custom [error interceptor](https://zod.dev/ERROR_HANDLING?id=customizing-errors-with-zoderrormap) where we could send analytics events.

Reply:
Sounds good 👍 my thinking is more for us to have a first warning that APIs are failing so we can expect people having UI issues as a result and avoid time chasing down this type of intermittent issue

Reply:
Oh yeah, we can probably centralize all API error tracking in general within the API layer.

We could use something like a shared Axios instance with a custom error interceptor to track failures. By having tracking a bit closer to source we could have much more granular event information (i.e. url, API name, http status, etc..)  instead of the top-level query key.

I just saw [this work](https://github.com/leather-io/extension/pull/5925) from @alter-eggo in `extension` - If we do any analytics here in `@leather.io/services` it may duplicate these existing API events. But the value prop seems high of getting more granular events data (e.g. 429 failures), so maybe we could use different event names for the query-level failures already in `extension` vs request-level failures. @alter-eggo @camerow thoughts?

Reply:
@alter-eggo sorry, that was bad phrasing on my part. I meant where the error is handled and analytics event sent rather than the where it originates from (i.e. RQ `onError  vs an Axios interceptor). My thought was that because `useQuery` can wrap higher-level calls that encapsulate >1 API request, and requests can be made imperatively outside of RQ, at some point it might make sense to move analytics into an Axios interceptor.

Anyway I see now that the error object in RQ `onError` already does have access to the undelrying response + status code, so its not a huge difference.